patch/src/lib.rs

```diff
@@ -555,13 +556,40 @@ fn pretty_sync_status(
     Ok(format!("ahead {}, behind {}", ahead, behind))
 }
 
+/// Make a human friendly string for commit version information.
+///
+/// For example '<oid> (branch1[, branch2])'.
+fn pretty_commit_version(
+    revision_oid: &git::Oid,
+    repo: &Option<git::Repository>,
+) -> anyhow::Result<String> {
+    let mut str = common::fmt::oid(revision_oid);
+    let mut branches: Vec<String> = vec![];
+
+    if let Some(repo) = repo {
+        for r in repo.references()?.flatten() {
+            if let (Some(oid), Some(name)) = (&r.target(), &r.shorthand()) {
+                if oid == revision_oid {
+                    branches.push(name.to_string());
+                };
```

Comment:
Ok one thing I noticed here, is I'm getting things other than branches match, eg. remote refs that aren't associated to local branches. So I think we need an additional check, maybe by trying to fetch the branch or something?

Reply:
Going to merge this already, can tackle it later.